### DISCUSSION TITLE
Send channel from Flutter app

### DISCUSSION THREAD

**Discussion by XxThunderBlastxX** [serverpod/serverpod#1198] [Q&A] [answered]
I am making a basic chat application where the users can chat in a single channel. I want to set and subscribe to the channel Id from my flutter app. How can I tell the server to listern to the channel ID I have set from my flutter app ??
In simple word I want to have multiple channels running on my serverpod server.

**Comment by vlidholt** [comment, marked as answer]
It's not an issue to have multiple channels in your server. If you are building a chat, a good starting point would be the [serverpod chat module](https://github.com/serverpod/serverpod/tree/main/modules/serverpod_chat). Even if you don't end up using the module itself, it's a good example on how to build a chat.

The shorter answer is that you want to pass the id of the channel to your server from your app. You can do this by creating a serializable class and passing it to the stream. Your endpoint would expect the first streaming message it receives to be a request to join a specific channel.

**Reply by XxThunderBlastxX** [reply]
Ok I will look the serverpod chat module. Thank you :)
